src/ucp/core/ucp_context.c

```diff
@@ -275,6 +298,21 @@ static ucs_status_t ucp_add_tl_resources(ucp_context_h context,
     return status;
 }
 
+static void ucp_check_unavailable_devices(str_names_array_t device_list, int *masks,
+                                     int index)
+{
+    int i;
+
+    if (masks[index] != -1) {
+        for (i = 0; i < device_list.count; i++) {
+            if (!(masks[index] & 1)) {
+                ucs_info("Device %s is not available", device_list.names[i]);
+            }
+            masks[index] >>= 1;
```

Comment:
it's not expected that a function which checks something about a parameter it receives will change it